impeller/entity/shaders/gaussian_blur/gaussian_blur.glsl

```diff
@@ -54,7 +55,7 @@ void main() {
   // just uses 0.5 as an optimization until the gaussian coefficients are
   // calculated and passed in from the cpu.
   for (float16_t i = -blur_info.blur_radius; i <= blur_info.blur_radius;
-       i += 2.0hf) {
+       i += blur_info.step_size) {
```

Comment:
This causes waffling that even more noticeable with the downscale approach.

Reply:
Interesting! I will dig into this afterwards. Its a nice win if we can get it working but is otherwise completely orthogonal to this change

Reply:
It's the issue I reported here: https://github.com/flutter/flutter/issues/138259